modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotRestoreWithIndexingTest.java

```diff
@@ -76,8 +76,11 @@ public void testBasicClusterSnapshotRestore() throws Exception {
 
         grid(0).snapshot().restoreSnapshot(SNAPSHOT_NAME, Collections.singleton(DEFAULT_CACHE_NAME)).get(TIMEOUT);
 
+        if (onlyPrimary)
+            awaitPartitionMapExchange();
```

Comment:
Why this is not necessary for regular snapshot?

Reply:
Because, for `onlyPrimary==false` snapshots all partitions are already there. Moreover index partition backed up too.
So, no additional actions required during restore.

Reply:
More detailed explanation required here.
Primary only case will restore only primary partitions on nodes and subsequent rebalance will be required?

Reply:
In `onlyPrimary` mode index rebuild will happen.
`awaitPartitionMapExchange` required to await while index rebuild ends so subsequent checks will pass.